src/plugins/intel_gpu/src/graph/input_layout.cpp

```diff
@@ -55,6 +55,13 @@ void input_layout_inst::set_data(memory::ptr mem) {
 
     _has_valid_input = true;
     _output_changed = true;
+    _shape_changed = mem->get_layout() != ol;
+
+    if (_shape_changed) {
+        node.invalidate_users();
```

Comment:
Actually this is no longer needed, because in dynamic shape scenario the actual shape will be propagated through kernel_impl_param, instead of the program_node's output layout.

Reply:
agree. removed it.